components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/StreamingListener.java

```diff
@@ -0,0 +1,215 @@
+        String queueGroup = natsProperties.getProperty(NatsConstants.NATS_STREAMING_QUEUE_GROUP);
+        boolean isManualAck = Boolean.parseBoolean(natsProperties.getProperty(NatsConstants.NATS_STREAMING_MANUAL_ACK));
+        String ackWait = natsProperties.getProperty(NatsConstants.NATS_STREAMING_ACK_WAIT);
+        String maxInFlight = natsProperties.getProperty(NatsConstants.NATS_STREAMING_MAX_IN_FLIGHT);
+        String subscriptionTimeout = natsProperties.getProperty(NatsConstants.NATS_STREAMING_SUBSCRIPTION_TIMEOUT);
+        String dispatcher = natsProperties.getProperty(NatsConstants.NATS_STREAMING_DISPATCHER);
+
+        if (StringUtils.isNotEmpty(durableName)) {
+            subscriptionOptions.durableName(durableName);
+        }
+
+        if (isManualAck) {
+            subscriptionOptions.manualAcks(); // if subscriptionOptions.manualAcks() is not set, it will auto ack
+        }
+
+        if (StringUtils.isNotEmpty(ackWait)) {
+            subscriptionOptions.ackWait(Duration.ofSeconds(Integer.parseInt(ackWait)));
+        }
+
+        if (StringUtils.isNotEmpty(maxInFlight)) {
+            subscriptionOptions.maxInFlight(Integer.parseInt(maxInFlight));
+        }
+
+        if (StringUtils.isNotEmpty(subscriptionTimeout)) {
+            subscriptionOptions.subscriptionTimeout(Duration.ofSeconds(Integer.parseInt(subscriptionTimeout)));
+        }
+
+        if (StringUtils.isNotEmpty(dispatcher)) {
+            subscriptionOptions.dispatcher(dispatcher);
+        }
+
+        subscription = connection.subscribe(subject, queueGroup, natsMessage -> {
+            String message = new String(natsMessage.getData());
+            printDebugLog("Message Received to NATS Inbound EP: " + message);
+            log.info("Message Received to NATS Inbound EP: " + message);
+            boolean isInjected = injectHandler.invoke(message.getBytes(), sequenceName, null, null);
+            // message is acknowledged only if the message is successfully injected to sequence (only for manual acks)
+            if (isInjected)
+                acknowledge(isManualAck, natsMessage);
+        }, subscriptionOptions.build());
+    }
+
+    /**
+     * Check if manual acks is set to true and ack manually.
+     *
+     * @param isManualAck boolean to enable manual ack.
+     * @param natsMessage the NATS message
+     */
+    private void acknowledge(boolean isManualAck, Message natsMessage) {
+        if (isManualAck) {
+            try {
+                natsMessage.ack();
+            } catch (IOException e) {
+                log.error("An error occurred while sending manual ack. Message might get redelivered. Sequence number: "
+                        + natsMessage.getSequence(), e);
+            }
+        }
+    }
+
+    /**
+     * Close the connection to NATS Streaming server and set connection to null.
+     */
+    @Override public void closeConnection() {
+        try {
+            if (subscription != null) subscription.close();
```

Comment:
Even though it is technically correct to have no braces for a single line statement inside, it is better if you can add braces for clarity.